### ISSUE TITLE
[M5] Continual language acquisition from lessons, demonstrations, corpora, and grounded interaction

### ISSUE THREAD

**Issue by SzeChunYiu** [SzeChunYiu/ORION-OCM#7]
# M5 — Continual language acquisition

**Parent:** #1  
**Depends on:** #6 (`M4_DIALOGUE_COGNITIVE_LOOP_GREEN`)  
**Required terminal:** `M5_CONTINUAL_LANGUAGE_LEARNING_GREEN`

## Goal

Demonstrate that OCM can **learn how to communicate** from multiple evidence channels rather than merely execute a fixed grammar or imitate text. The machine must acquire, revise, transfer, and revoke linguistic knowledge and communication procedures while preserving prior competence.

Language is treated as the first full Machine Epistemics learning ladder:

```text
learn linguistic knowledge
→ learn communication procedures
→ learn language-learning strategies
→ revise language representations
→ improve communication organization
```

M5 focuses on the first three layers and records representation failures that may later justify changes.

---

## 1. Freeze acquisition regimes separately

Run these regimes independently before mixing them.

### E0 — explicit lessons

Examples:

- dictionary entries;
- grammar explanations;
- morphology rules;
- explicit corrections;
- construction descriptions;
- pragmatic/discourse instructions.

Measure how quickly explicit teaching changes protected behavior.

### E1 — aligned meaning↔utterance demonstrations

Inputs contain a known semantic/task state plus a human utterance. Use these to learn form↔meaning correspondences without requiring explicit grammar terminology.

### E2 — raw corpora/books/articles

Only surface text is visible. The learner may infer:

- recurring forms;
- distributional clusters;
- morphology;
- lexical associations;
- construction candidates;
- collocations;
- discourse/style regularities.

It may **not** treat distributional regularity as grounded world truth.

### E3 — grounded interaction

The machine interacts with a controlled environment/user and receives observations/outcomes. It must connect utterances to actions, referents, corrections, and communicative success/failure.

### E4 — mixed curriculum

Only after E0–E3 are individually characterized. Measure complementarity rather than assuming “more data is better.”

---

## 2. Core learned objects

M5 must be able to acquire/revise at least:

```text
Lexeme / lexical sense
Morphological pattern
Construction / syntax-semantic mapping
Discourse convention
Reference convention
Idiomatic / multiword construction
Register/style pattern
Communication procedure
Language-learning strategy record
```

Every learned object carries provenance, evidence channel, scope, dependencies, known counterexamples, and revision lineage.

---

## 3. Learning a new word

Required curriculum:

1. teach a concrete noun from one grounded/aligned example;
2. use it in unseen grammatical contexts;
3. teach a second sense of the same surface word;
4. test context-sensitive disambiguation;
5. revoke the first sense evidence;
6. ensure the second sense remains;
7. relearn the first from a new source.

Metrics:

- examples/interactions to threshold competence;
- compositional reuse;
- sense interference;
- revocation locality;
- retained unrelated vocabulary.

---

## 4. Learning a new construction

Protected post-deployment test:

1. freeze the initial language system;
2. introduce a construction absent from training;
3. teach by explicit lesson vs demonstrations vs interaction;
4. test unseen lexical fillers and semantic combinations;
5. introduce exceptions/near-neighbor constructions;
6. measure whether the learner overgeneralizes;
7. revoke or revise the lesson and verify local correction.

This is a required Alpha discriminator. A system that only works for constructions present in its initial corpus does not pass.

---

## 5. Learning morphology and exceptions

Compare strategies for productive pattern induction:

- rule induction;
- analogy/nearest paradigm;
- probabilistic pattern;
- hybrid rule + exceptions;
- learned operator over forms.

Use protected novel stems and irregular cases.

Record when a general rule should split/specialize because exceptions accumulate.

---

## 6. Learning from raw text without semantic laundering

Implement a two-stage protocol:

```text
raw text → ungrounded form/distribution hypotheses
aligned/grounded evidence → semantic binding and warrant upgrade
```

Required status labels:

```text
UNGROUNDED_FORM_ONLY
CANDIDATE_SEMANTIC_BINDING
GROUNDED_CONSTRUCTION
CONTRADICTED
REVOKED
```

Hostile: a corpus pattern that correlates with a protected false meaning must not become grounded merely because it is frequent.

---

## 7. Books/articles as learning data

Create a legal/data-governed book/article pipeline.

Preferred initial sources:

- public-domain English texts (e.g. Project Gutenberg-compatible public-domain works, after exact license/source validation);
- open web/news/article corpora only where redistribution/use terms are compatible;
- BabyLM official corpora under the challenge's current terms.

For every text source record:

```text
title/source
edition/version
license/public-domain basis
acquisition date
hash
word count
preprocessing
whether held-out evaluation text overlaps
```

Books should primarily test acquisition of form, style, discourse, rare constructions, and lexical distribution. Factual claims in books are not automatically trusted knowledge.

---

## 8. BabyLM 2026 sample-efficiency programme

Use current BabyLM rules as a serious external sample-efficiency arena.

At execution time pin the official release and evaluation code. Current 2026 tracks provide:

- Strict: ≤100M words;
- Strict-Small: ≤10M words;
- multilingual track with focus including English, Dutch, and Chinese;
- multimodal/interactive/teacher-feedback approaches permitted under current Strict constraints.

OCM should run at least:

```text
1M exploratory budget (internal)
10M Strict-Small-aligned budget
100M Strict-aligned budget if computationally feasible
```

The mechanism arm must disclose every word/lesson/annotation/input channel; aligned semantic supervision counts as additional information and must not be compared naively with text-only parents.

---

## 9. CHILDES / developmental dialogue study

Use CHILDES only under current TalkBank access terms. Record whether selected corpora are open, registration-required, or controlled.

Purpose:

- measure learning from realistic caregiver/child dialogue patterns;
- compare order/curriculum effects;
- study incremental construction/vocabulary acquisition;
- test reference/repair/interaction patterns.

Do not claim biological plausibility merely because CHILDES is used.

If registration/licensing cannot be satisfied in CI, keep acquisition scripts/manifest and report `CANNOT_CHECK_CHILDES_DATA` rather than copying restricted data.

---

## 10. Curriculum learning experiments

Freeze and compare curriculum orders, for example:

```text
raw text → aligned demos → interaction
explicit lessons → interaction → raw text
interaction-first
mixed shuffled
adaptive OCM-selected curriculum
```

Measure:

- final competence;
- data/sample cost;
- number of active constructions/lexemes;
- catastrophic/negative transfer;
- correction cost;
- training/update compute.

A learned curriculum policy must be compared against fixed curricula on unseen learning tasks.

---

## 11. Active language learning

Allow OCM to choose information-seeking actions such as:

- ask meaning of unknown word;
- request paraphrase;
- ask which referent is intended;
- ask for another example;
- propose interpretation and request confirmation;
- test a construction in a safe grounded task.

Use value-of-information/resource-aware selection.

Metrics:

- competence per question/intervention;
- unnecessary questions;
- oracle regret in controlled environments;
- total interaction burden.

---

## 12. Negative transfer and overgeneralization

Required protected families:

- similar surface construction with different meaning;
- English word order transferred incorrectly to SOV-like mini-language;
- regular morphology applied to irregular form;
- domain-specific sense transferred to general context;
- stylistic pattern treated as grammar rule;
- proverb/idiom interpreted compositionally when idiomatic;
- high-frequency corpus error or quoted false statement.

Measure transfer precision:

```text
P(transfer beneficial | OCM chose transfer)
```

and harmful-transfer rate separately.

---

## 13. Multilingual preflight

Do **not** attempt full multilingual fluency before English Alpha. But M5 must include one small protected transfer study showing that the architecture does not constitutionally assume English SVO.

Use a controlled mini-language or a small licensed corpus in Chinese/Dutch from the current BabyLM multilingual setup if feasible.

Questions:

- which learned language objects transfer safely?
- can semantic structures transfer while word order does not?
- can morphology/lexicon remain language-scoped?
- does the system detect rather than force incompatible syntax?

Terminal here is a transfer diagnostic, not multilingual Alpha.

---

## 14. Retention and continual-learning protocol

After each learning episode evaluate a frozen retention set.

Track:

```text
new capability gain
old capability loss
unrelated capability change
update compute
memory growth
number of reopened objects
```

Require local revocation tests after at least:

- lexical update;
- morphology update;
- construction update;
- discourse rule update.

No “retrain from scratch” may count as local continual learning for the principal OCM claim, though it can be a comparator.

---

## 15. Metrics

Primary metric vector:

```text
sample efficiency
held-out semantic correctness
grammaticality
compositional generalization
post-deployment learning success
retention / forgetting
negative-transfer harm
transfer precision
calibration/unknown behavior
revocation locality
update wall-time/compute/memory
interaction/question count
```

Report learning curves, not just endpoint accuracy.

A useful derived measure:

```text
LifetimeLearningEfficiency
 = area under competence-vs-cumulative-information curve
```

but never replace the raw vector with this single scalar.

---

## 16. Comparators

Under declared matched regimes compare against:

- n-gram/statistical language learner where informative;
- recurrent/Transformer model trained on the same 10M/100M text budget;
- grammar/construction induction parent appropriate to the controlled dataset;
- retrieval/memorization baseline;
- fine-tuning/retraining baseline for post-deployment lessons;
- OCM ablations: no warrant, no active learning, no local revocation, no construction abstraction.

If a parent receives only raw text while OCM receives aligned meanings/teacher feedback, report that information difference explicitly and do not call the comparison matched.

---

## 17. Hostile tests

Required:

- protected examples leak into training corpus;
- teacher-generated annotation contains answer labels unavailable to comparator;
- raw corpus miner calls external LLM for semantics;
- exception stored but general rule still overrides it;
- revoked lesson remains active via compiled macro;
- new construction learning destroys unrelated construction;
- corpus frequency promoted to world truth;
- active learner asks the gold label directly when task rules prohibit it;
- curriculum chosen after inspecting protected outcomes;
- multilingual transfer reuses English word order despite counterevidence;
- retrain-from-scratch mislabeled as local update.

---

## 18. Deliverables

- [ ] unified language learner interface
- [ ] explicit lesson learner
- [ ] aligned demonstration learner
- [ ] raw corpus/form induction learner
- [ ] grounded interaction learner
- [ ] active learning policy/reference
- [ ] vocabulary/sense acquisition lifecycle
- [ ] construction acquisition lifecycle
- [ ] morphology/exception lifecycle
- [ ] BabyLM data/evaluation adapter + exact manifest
- [ ] public-domain book/article manifest and loader
- [ ] CHILDES acquisition manifest/adapters if authorized
- [ ] multilingual preflight
- [ ] retention/negative-transfer suite
- [ ] matched comparator training/evaluation scripts
- [ ] M5 receipt with full information/resource accounting

---

## 19. Exit gate

Close only when OCM can, after the initial system is frozen:

- learn new words and at least one genuinely new construction from small post-deployment evidence;
- reuse them compositionally in protected contexts;
- learn from at least explicit lesson, aligned demonstration, raw corpus, and grounded interaction regimes with their contributions measured separately;
- preserve unrelated prior language ability through updates/revocation;
- detect/refuse at least the registered harmful-transfer cases;
- complete the frozen 10M-word sample-efficiency experiment or report an explicit computational/data `CANNOT_CHECK`;
- pass all required data-leakage and semantic-laundering hostiles;
- report matched parent learning curves and resource vectors.

Terminal:

```text
M5_CONTINUAL_LANGUAGE_LEARNING_GREEN
```

Fluent output produced only after static pretraining does not satisfy this milestone.

**Comment by SzeChunYiu** [comment]
## Tiny-prototype data-budget amendment

For `LANGUAGE_KSO_ALPHA`, the primary completion gate is **small-data continual learning**, not a 100M-word language-model training run.

Freeze a staged budget such as:

```text
10^3–10^4 words/examples: mechanism/debug regime
10^4–10^5: protected tiny-language regime
10^5–10^6: primary scaling regime if feasible
10M BabyLM-style: secondary external scaling reference
100M: optional future scale study, not an Alpha blocker
```

Exact thresholds should be frozen after M3/M4 dev calibration, before protected M5 outcomes.

The key claim is whether OCM can learn a new word/construction with **small local information/update cost**, reuse it compositionally, retain unrelated competence, and later learn related structures more cheaply because reusable structure already exists.

Track separately:

```text
initial acquisition information/cost
new-capability information/cost
local correction/revocation cost
retention cost
inference/navigation cost
memory/index growth
```

Compare against small matched neural/grammar/memory parents trained or adapted on the **same information channels**. BabyLM remains useful context, but Alpha does not wait for frontier-scale fluency.

**Comment by SzeChunYiu** [comment]
M5 build is up as a PR (branch m5/continual-acquisition): lexical/sense lifecycle by exact alignment, morphology strategies with exceptions and split recommendation, raw-corpus form induction without semantic laundering, grounded-interaction learner over registered outcome functions, active learning by expected elimination (gold label prohibited), transfer decisions + retention vector + SOV preflight; regime-separated receipt (E0 134/134 with 6 lessons; E1 115/134 with 7 demonstrations; E2 semantic gain 0 with 88 ungrounded forms; E3 37/134 from one interaction; four curricula; retention old loss 0/22). Custody: Gutenberg acquired; BabyLM CANNOT_CHECK_BABYLM_DATA. Spec OCM_ACQUISITION_V1; registry KS-T51…T59 (T59 BabyLM/CHILDES/comparators open). Ledger S17.

**Comment by SzeChunYiu** [comment]
M5_CONTINUAL_LANGUAGE_LEARNING_GREEN (microworld scope) — PR #20 merged with a merge commit (2e4bca0). Receipt docs/provenance/M5_RECEIPT_V1.json; CI m5-acquisition green on the merged head (M0–M4 also green). Regimes measured separately: E0 134/134 (6 lessons), E1 115/134 (7 demonstrations), E2 semantic gain 0 (88 ungrounded forms), E3 37/134 (1 interaction), four fixed curricula; retention old loss 0/22; SOV preflight detected not forced. Open (KS-T59): BabyLM 10M run (CANNOT_CHECK_BABYLM_DATA until the release/terms are pinned), CHILDES adapters, matched comparator curves — carried into M7. Closing at microworld scope; the exit gate's 10M-word experiment is reported as an explicit CANNOT_CHECK.
